File: server/php/src/Application/DTOs/Payment/ResponseDto.php
<?php

namespace OnlinePayments\ExampleApp\Application\DTOs\Payment;

use OnlinePayments\ExampleApp\Application\Domain\Enums\AdditionalPaymentActions\Status;
use OnlinePayments\ExampleApp\Application\Domain\Payments\StatusOutput;

class ResponseDto
{
    public ?string $id;
    public ?Status $status;
    public ?StatusOutput $statusOutput;

    public function __construct(
        ?string $id = null,
        ?Status $status = null,
        ?StatusOutput $statusOutput = null
    ) {
        $this->id = $id;
        $this->status = $status;
        $this->statusOutput = $statusOutput ?? new StatusOutput();
    }
}

File: server/ruby/app/business/dtos/additional_payment_actions/response_dto.rb
module Business
  module Dtos
    module AdditionalPaymentActions
      class ResponseDto
        attr_accessor :id, :status, :status_output

        def initialize(id: nil, status: nil, status_output: nil)
          @id = id
          @status = status
          @status_output = status_output
        end
      end
    end
  end
end